crates/telio-firewall/src/firewall.rs

```diff
@@ -1401,7 +1501,195 @@ pub mod tests {
 
     #[rustfmt::skip]
     #[test]
-    fn firewall_icmp() {
+    fn firewall_icmp_cache_general() {
+        struct TestInput {
+            src1: &'static str,
+            src2: &'static str,
+            src3: &'static str,
+            dst: &'static str,
+            make_icmp: MakeIcmp,
+        }
+
+        let test_inputs = vec![
+            TestInput { src1: "8.8.8.8",              src2: "8.8.4.4",              src3: "4.4.4.4",              dst: "127.0.0.1", make_icmp: &make_icmp4 },
+            TestInput { src1: "2001:4860:4860::8888", src2: "2001:4860:4860::8844", src3: "2001:4860:4860::4444", dst: "::1",       make_icmp: &make_icmp6 },
+        ];
+        for TestInput{ src1, src2, src3, dst, make_icmp } in test_inputs {
+            let make_packet = |src, dst, icmp_type, id, seq| -> Vec<u8> {
```

Comment:
this helper is copied 3 times. I think it can be extracted to the separate function